packages/core/src/services/providerApiKeys/destroy.ts

```diff
@@ -1,21 +1,42 @@
 import { eq } from 'drizzle-orm'
+import { DatabaseError } from 'pg'
 
 import { ProviderApiKey } from '../../browser'
 import { database } from '../../client'
-import { Result, Transaction } from '../../lib'
+import {
+  BadRequestError,
+  databaseErrorCodes,
+  Result,
+  Transaction,
+} from '../../lib'
 import { providerApiKeys } from '../../schema'
 
 export function destroyProviderApiKey(
   providerApiKey: ProviderApiKey,
   db = database,
 ) {
   return Transaction.call(async (tx) => {
-    const result = await tx
-      .delete(providerApiKeys)
-      .where(eq(providerApiKeys.id, providerApiKey.id))
-      .returning()
-    const deleted = result[0]
+    try {
+      const result = await tx
+        .delete(providerApiKeys)
+        .where(eq(providerApiKeys.id, providerApiKey.id))
+        .returning()
+      const deleted = result[0]
 
-    return Result.ok(deleted)
+      return Result.ok(deleted)
+    } catch (error) {
+      if (
+        error instanceof DatabaseError &&
+        error.code === databaseErrorCodes.foreignKeyViolation
+      ) {
```

Comment:
This is not the right way of handle this. We don't have to wait to arrive to the DB to fail. This has to be validated in the action with a zod async validation. 

Example:
https://github.com/latitude-dev/latitude-llm/blob/main/apps/web/src/actions/user/setupAction.ts#L22-L25

Reply:
This is like if do a catch in a Rails transaction. We we would do that having models and validations?

Reply:
Please change it to don't set a bad precent

Reply:
i disagree why add an extra db roundtrip to all requests for a validation when you can just let the db validation fail and return a nicely formatted error 🤷🏼

Reply:
i absolutely do not think this is a bad precedent

Reply:
But this is exactly how Rails works. Is a validation check againts Db. Why matters the roundtrip in a POST operation? I don't think is the bottle neck.

Also services should receive valid data the same way they receive valid scopes.

Having to deal with try/catch feels nasty. Having zod which provides a nice layer between client and server

Reply:
Any way. I said my part feel free to ignore